tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -255,13 +257,51 @@ public function testFetchAllWithMissingTypes(): void
             $this->markTestSkipped('mysqli and sqlsrv actually supports this');
         }
 
+        if (
+            $this->connection->getDriver() instanceof DB2Driver
+        ) {
+            $this->markTestSkipped(
+                'ibm_ibm2 may or may not report the error depending on the PHP version and the connection state'
+            );
+        }
+
         $datetimeString = '2010-01-01 10:10:10';
         $datetime       = new DateTime($datetimeString);
         $sql            = 'SELECT test_int, test_datetime FROM fetch_table WHERE test_int = ? AND test_datetime = ?';
 
         $this->expectException(DBALException::class);
 
-        $this->connection->fetchAll($sql, [1, $datetime]);
+        $fetch($this->connection, $sql, [1, $datetime]);
+    }
+
+    /**
+     * @return iterable<string,array<int,mixed>>
```

Comment:
```suggestion
     * @return iterable<string,array{0:callable}>
```

Reply:
We're getting into a danger zone here. Techhically, it's something like `iterable<string,array{0:callable(string,<array<int|string,mixed>,<array<int|string,int|string|Type|null>>)}>`. Not sure how specific we want to be here.

Reply:
Ugh yeah let's keep it as is.